src/highlight/HighlightAnnotations.tsx

```diff
@@ -1,46 +1,137 @@
 import * as React from 'react';
+import noop from 'lodash/noop';
 import HighlightCanvas from './HighlightCanvas';
 import HighlightCreator from './HighlightCreator';
 import HighlightList from './HighlightList';
+import HighlightSvg from './HighlightSvg';
+import HighlightTarget from './HighlightTarget';
+import PopupReply from '../components/Popups/PopupReply';
 import { AnnotationHighlight } from '../@types';
-import { isValidHighlight, sortHighlight } from './highlightUtil';
-
+import { CreatorHighlight, CreatorStatus, Mode } from '../store';
 import './HighlightAnnotations.scss';
 
+export interface CreateArg extends Pick<AnnotationHighlight, 'target'> {
+    message: string;
+}
+
 type Props = {
     activeAnnotationId: string | null;
     annotations: AnnotationHighlight[];
+    createHighlight?: (arg: CreateArg) => void;
     isCreating: boolean;
+    location: number;
+    message: string;
     setActiveAnnotationId: (annotationId: string | null) => void;
+    setMessage: (message: string) => void;
+    setMode: (mode: Mode) => void;
+    setStaged: (staged: CreatorHighlight | null) => void;
+    setStatus: (status: CreatorStatus) => void;
+    staged?: CreatorHighlight | null;
+    status: CreatorStatus;
 };
 
-export default class HighlightAnnotations extends React.PureComponent<Props> {
-    static defaultProps = {
-        annotations: [],
-        isCreating: false,
-    };
+const HighlightAnnotations = (props: Props): JSX.Element => {
+    const {
+        activeAnnotationId,
+        annotations = [],
+        createHighlight = noop,
+        isCreating = false,
+        location,
+        message,
+        setActiveAnnotationId,
+        setMessage,
+        setMode,
+        setStaged,
+        setStatus,
+        staged,
+        status,
+    } = props;
+    const [highlightRef, setHighlightRef] = React.useState<HTMLAnchorElement | null>(null);
 
-    handleAnnotationActive = (annotationId: string | null): void => {
-        const { setActiveAnnotationId } = this.props;
+    const canReply = status !== CreatorStatus.started && status !== CreatorStatus.init;
+    const isPending = status === CreatorStatus.pending;
 
+    const handleAnnotationActive = (annotationId: string | null): void => {
         setActiveAnnotationId(annotationId);
     };
 
-    render(): JSX.Element {
-        const { activeAnnotationId, annotations, isCreating } = this.props;
-        const sortedAnnotations = annotations.filter(isValidHighlight).sort(sortHighlight);
+    const handleCancel = (): void => {
+        setMessage('');
```

Comment:
Should we add a `resetCreator` action and use it in both highlights and regions?